File: 0001-two-sum/0001-two-sum.c

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* result = (int*)malloc(2*sizeof(int));
    int i,j;
    for(i=0;i<numsSize;i++)
    {
        for(j=i+1;j<numsSize;j++)
        {
            if(nums[i]+nums[j]==target)
            {
                result[0] = i;
                result[1] = j;
                *returnSize=2;
                return result;
            }
        }
       
    }
     *returnSize=0;
        return NULL;
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        int i;
        hm.put(nums[0],0);
        for(i=1;i<nums.length;i++)
        {
            int tar = target - nums[i];
            if(hm.containsKey(tar))
            {
                return new int[]{hm.get(tar),i};
            }
            hm.put(nums[i],i);
        }
        return new int[]{0,0};
    }
}